codegenerator/cli/src/cli_args/interactive_init/evm_prompts.rs

```diff
@@ -252,13 +254,41 @@ fn get_converter_network_u64(
                 Some(r) => r.clone(),
                 None => prompt_for_rpc_url()?,
             };
-            converters::NetworkKind::Unsupported(network_id, rpc_url)
+            let start_block = match start_block {
+                Some(s) => s.clone(),
+                None => prompt_for_start_block()?,
+            };
+
+            converters::NetworkKind::Unsupported {
+                network_id,
+                rpc_url,
+                start_block,
+            }
         }
     };
 
     Ok(network)
 }
 
+///Prompt the user to enter a starting block
+///only prompt when used when using rpc as it could
+///be very slow to have the startblock at 0 with rpc 🦶🔫
+fn prompt_for_start_block() -> Result<u64> {
+    let prompt = Text::new(
+        "You have entered a network that is unsupported by HyperSync. Please provide a start block for this network \
+            (this can be edited later in config.yaml):",
+    )
+    .prompt();
+
+    let start_block: String = prompt.context("Failed during start block prompt")?;
+    let start_block: u64 = start_block
+        .trim()
+        .parse()
+        .context("Failed to parse start block, start block must be a number")?;
+
+    Ok(start_block)
+}
```

Comment:
@DenhamPreen have a look at inquires docs on "CustomType" https://github.com/mikaelmello/inquire?tab=readme-ov-file#customtype

It's not a big one but the parsing then becomes built in and on top of that you get an error message where you can retry rather than a failure during the prompting process.